handler/src/main/java/io/netty/handler/ssl/SignatureAlgorithmConverter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.ssl;
+
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Converts OpenSSL signature Algorithm names to
+ * <a href="https://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html#Signature">
+ *     Java signature Algorithm names</a>.
+ */
+final class SignatureAlgorithmConverter {
+
+    private SignatureAlgorithmConverter() { }
+
+    // OpenSSL has 3 different formats it uses at the moment we will match against all of these.
```

Comment:
Why not just enumerate them in the regex then?

Reply:
@carl-mastrangelo not sure I understand... Can you show me some code ?

Reply:
Make the regex `"^(SHA256withECDSA|SHA512withRSA|SHA256withDSA)$"`

Reply:
Ah ok... maybe the comment is misleading. It supports more then 3 algorithms. It has just 3 different way to format the supported algorithms

Reply:
I guess the comment is correct - it says 'formats' which is not 'algorithms', but it's probably better giving some examples in the comment.

Reply:
done